src/test/java/org/minions/devfund/yuri/linkedlist/ContainsTest.java

```diff
@@ -0,0 +1,346 @@
+package org.minions.devfund.yuri.linkedlist;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.util.LinkedList;
+
+/**
+ * Source code recreated from a .class file by IntelliJ IDEA.
+ * (powered by Fernflower decompiler)
+ */
+public class ContainsTest {
+    private static final int LIST1FIRSTVALUE = 4;
+    private static final int LIST1SECONDVALUE = 8;
+    private static final int LIST1THIRDVALUE = 3;
+    private static final int LIST1FOURTHVALUE = 12;
+    private static final int LIST1FIFTHVALUE = 6;
+    private static final int LIST2EXTRAVALUE1 = 5;
+    private static final int LIST2EXTRAVALUE2 = 7;
+    private static final int LIST2EXTRAVALUE3 = 9;
+    private static final int LIST2EXTRAVALUE4 = 2;
+    private static final String MSG1 = "containsSubsequence returns true when first LinkedList contains all elements";
+    private static final String MSG2 = "when first LinkedList contains second";
+    private static final String MSG3 = "containsSubsequence throws";
+    public static final String CONTAINS_SUBSEQUENCE = "containsSubsequence returns false ";
+
+    /**
+     * test.
+     */
+    @Test
+    public void testContainsMiddle() {
+        LinkedList var1 = new LinkedList();
+        LinkedList var2 = new LinkedList();
+        var1.add(LIST1FIRSTVALUE);
+        var1.addLast(LIST1SECONDVALUE);
+        var1.addLast(LIST1THIRDVALUE);
+        var1.addLast(LIST1FOURTHVALUE);
+        var1.addLast(LIST1FIFTHVALUE);
+        var2.add(LIST1FIRSTVALUE);
+        var2.add(LIST1SECONDVALUE);
+        var2.add(LIST1THIRDVALUE);
+
+        try {
+            boolean var3 = LinkedListUtils.containsSubsequence(var1, var2);
+            Assert.assertTrue(CONTAINS_SUBSEQUENCE + MSG2 + " in "
+                    + "middle of list", var3);
+        } catch (Exception var4) {
+            Assert.fail(MSG3 + var4 + MSG2 + " in middle of list");
+        }
+
+    }
+
+    /**
+     * test.
+     */
+    @Test
+    public void testContainsFront() {
+        LinkedList var1 = new LinkedList();
+        LinkedList var2 = new LinkedList();
+        var1.add(LIST1FIRSTVALUE);
+        var1.addLast(LIST1SECONDVALUE);
+        var1.addLast(LIST1THIRDVALUE);
+        var1.addLast(LIST1FOURTHVALUE);
+        var1.addLast(LIST1FIFTHVALUE);
+        var2.add(LIST1FIRSTVALUE);
+        var2.add(LIST1SECONDVALUE);
+        var2.add(LIST1THIRDVALUE);
+
+        try {
+            boolean var3 = LinkedListUtils.containsSubsequence(var1, var2);
+            Assert.assertTrue("containsSubsequence returns false " + MSG2 + " at front of list", var3);
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Use already-defined constant 'CONTAINS_SUBSEQUENCE' instead of duplicating its value here. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)